### DISCUSSION TITLE
Send Rich Text Field to REST API

### DISCUSSION THREAD

**Discussion by SantoshSrinivas79** [Budibase/budibase#4311] [Help]
Where I sent a rich text field value to a REST POST API, I am getting the following error. The REST call is not even executing ... so the error is thrown withing Budibase

`Invalid JSON for request body`

**Comment by mjashanks** [comment]
Hey @SantoshSrinivas79 

I was unable to reproduce this, my steps
1. Added REST API datasource, with parameter
2. On a screen, added a Form + Rich text
3. Inside the form, added a Datasource, connected to my query
4. Passed the value from the rich text to the query, via the query's parameter.

All worked ok. Could you give a bit more detail?

**Reply by SantoshSrinivas79** [reply]
@mjashanks sorry ... forgot to mention the important part ... the calls don't work when the input in the text box has line breaks (enter keys)

**Reply by mjashanks** [reply]
Mine seems to still work. The eventual API request fails because the API doesn't like it. But budibase handles it as expected...

![image](https://user-images.githubusercontent.com/3524181/152609711-0421415e-72b6-4bd6-b31e-00b85999d6ee.png)

![image](https://user-images.githubusercontent.com/3524181/152609777-e4f84cb7-3adb-4545-bfe9-edd6b604aa32.png)

**Reply by SantoshSrinivas79** [reply]
Hi @mjashanks ... I see to be missing something here .. could you pls advise.
attached is the project export too

<img width="1440" alt="Screenshot 2022-02-05 at 11 57 09 PM" src="https://user-images.githubusercontent.com/1036163/152654357-00ba4cca-5eee-4623-a69b-3c668ab842be.png">

[TestRichText-export-1644085701154.txt](https://github.com/Budibase/budibase/files/8008632/TestRichText-export-1644085701154.txt)

**Reply by SantoshSrinivas79** [reply]
@mjashanks just tried again ... this seems to be break when I use binding with `raw (JSON)` for creating the form body

**Reply by mjashanks** [reply]
Got it - thanks for that.

So - this is a bug, which I shall log.

The workaround is as follows
1. In your `Execute Query` action, pass `{{ JSONstringify New Form.Fields.test }}` as a parameter. This will escape the string properly.
2. In your JSON body, remove the quotes...

so: `"test":"{{test}}"` becomes `"test":{{test}}` - this is because `JSONstringify` adds it's own quotes.
